tools/installer/lib/ide-setup.js

```diff
@@ -93,6 +97,643 @@ class IdeSetup extends BaseIdeSetup {
     }
   }
 
+  async setupOpenCode(installDir, selectedAgent, spinner = null, preConfiguredSettings = null) {
```

Comment:
Ok I know this is the same style that seems to be prevalent in the codebase, but reading that 600 lines function is giving me headaches. 😆